packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -181,11 +181,11 @@ class RequestManager {
       account: account,
       cacheKey: uri.toString(),
       getCacheParameters: () async {
-        final response = await account.client.executeRawRequest(
-          'HEAD',
-          uri,
-          headers: headers,
-        );
+        final request = http.Request('HEAD', uri);
+        if (headers != null) {
+          request.headers.addAll(headers);
+        }
+        final response = await account.client.send(request);
 
         return CacheParameters.parseHeaders(response.headers);
       },
```

Comment:
```suggestion
        final response = await account.client.head(
          uri,
          headers: headers,
        );

        return CacheParameters.parseHeaders(response.headers);
      },
```

There is no benefit over your proposed changes but I find this more readable.